app/controllers/reorder_announcements_controller.rb

```diff
@@ -14,21 +14,17 @@ def update
     end
 
     if draft_updater.send
-      redirect_to coronavirus_page_path(slug), notice: "Announcements were successfully reordered."
+      redirect_to coronavirus_page_path(coronavirus_page.slug), notice: "Announcements were successfully reordered."
     else
       message = "Sorry! Announcements have not been reordered: #{draft_updater.errors.to_sentence}."
-      redirect_to reorder_coronavirus_page_announcements_path(slug), alert: message
+      redirect_to reorder_coronavirus_page_announcements_path(coronavirus_page.slug), alert: message
     end
   end
 
 private
 
-  def slug
-    params[:slug] || params[:coronavirus_page_slug]
-  end
-
   def coronavirus_page
-    @coronavirus_page ||= CoronavirusPages::ModelBuilder.call(slug)
+    @coronavirus_page ||= CoronavirusPage.find_by(slug: params[:coronavirus_page_slug])
   end
```

Comment:
This looks a nice simplification 👍 

One thing I notice that is different is that model builder does a find_or_create_by whereas this only does a find_by, so I guess this can crash if the model doesn't exist? Perhaps you just need to use `find_by!` so a 404 is returned?

I experimented by putting in a random slug into integration and that generated a 500 so I don't think the ModelBuilder approach is working correctly: https://collections-publisher.integration.publishing.service.gov.uk/coronavirus/asdfasdf/sub_sections/new

Naturally this comment also applies to the two other similar controllers.